### PR TITLE
Feature/adding node refresh

### PR DESCRIPTION
**Pull request by sjoedwards**
Adding an automatic build procedure for our React, just need to npm install and run 'gulp develop'

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    var stream =
+        gulp.src(['src/index.jsx'])
+        .pipe(named())
+        .pipe(plumber())
+        .pipe(webpack( config ))
+        .pipe(gulp.dest("build/src"));
+    return stream;
+})
+
+gulp.task('BSRefresh', function (done) {
```
Comment (chrisgrounds) [inline gulpfile.js]: Is this duplicated by the compile-watch task or is it just useful for something else?

**Review by chrisgrounds — APPROVED** [review]
Damn this is nice, Sam!

NB. FYI, on Ubuntu, I had to change the browser name from "google chrome" to "google-chrome" to get it to load the browser. I guess it is just named differently from OS to OS.

File: gulpfile.js (hunk 1 above)
Comment (sjoedwards) [reply gulpfile.js]: @chris-bacon Good spot, this is duplicate code!

**Comment by sjoedwards** [comment]
@chris-bacon Weird about Ubuntu. Theres probably a polyfill for that somewhere, I'll add it to the todo.

File: gulpfile.js (hunk 1 above)
Comment (sjoedwards) [reply gulpfile.js]: Pushed to remove redundant code

**Review by adamllwilliams — CHANGES_REQUESTED** [review]
Was going to approve but apparently I'm building emails. It's a no from me.

**Review by adamllwilliams — APPROVED** [review]

**Review by merrett010 — APPROVED** [review]
Vary nais -- think there's some merge conflicts but I've read through and pretty much all of it makes sense. 

However, would love a quick 2 min chat with someone about how some of the socket stuff works in backend.js and how 
import io from 'socket.io-client' 
in the App actually imports anything as I couldnt work out where that imports stuff from.

Well done guys!

**Comment by sjoedwards** [comment]
Cheers for approving guys, I'll look at the mergy wergy conflict

**Comment by chrisgrounds** [comment]
> However, would love a quick 2 min chat with someone about how some of the socket stuff works in backend.js and how `import io from 'socket.io-client'` in the App actually imports anything as I couldnt work out where that imports stuff from.

@merrett010 Don't know if this is what you mean, but `socket.io-client` seems to be exposed by `socket.io`, so it will pull it out of node_modules (when that's installed). Is that what you mean?

@sjoedwards 10/10 get this bad boy merged...

**Comment by merrett010** [comment]
@chris-bacon Oooohhh okay yeah that makes sense if it's in node modules. Was looking around non-git-ignored files for ages. cheers that makes sense

**Pull request merged**